src/screens/MoreScreen.js

```diff
@@ -1,33 +1,95 @@
+import {AppLoading} from 'expo';
 import React, {Component} from 'react';
-import {Text} from 'react-native';
-import {Icon} from 'react-native-elements';
+import {Platform, Text, View} from 'react-native';
+import {connect} from 'react-redux';
+import {Icon, Button, Card} from 'react-native-elements';
 
-import {CardSection, Card} from '../components/common/index';
+import {logUserOutOfFacebook} from '../actions';
+
+import {more_screen} from '../locale.en';
+import {navKeys} from '../constants';
 
 class MoreScreen extends Component {
+    componentDidMount() {
+        const {token, navigation: {navigate, state: {key}}} = this.props;
+
+        if (!token && key !== navKeys.LOGIN) {
+            navigate(navKeys.LOGIN);
+        }
+
+        this.props.navigation.setParams({onLogout: this.onLogout});
+    }
+
+    componentWillReceiveProps(nextProps) {
+        const {token, navigation: {navigate, state: {key}}} = nextProps;
+
+        if (!token) {
+            navigate(navKeys.LOGIN);
+        }
+    }
+
+    onLogout = () => {
+        this.props.logUserOutOfFacebook();
+    };
+
+    static renderRightNavButton = ({onLogout}) => {
+        if (Platform.OS === 'ios') {
+            return (
+                <Button
+                    title={more_screen.right_nav}
+                    onPress={onLogout}
+                    backgroundColor="rgba(0,0,0,0)"
+                    color="rgba(0,122,255,1)"
+                />
+            );
+        } else if (Platform.OS === 'android') {
```

Comment:
What I'd mentioned above; this is a static `renderRightNavButton` but we pass it a callback of the method `onLogout` so that it can use the method's instance function.